yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnSparkHadoopUtil.scala

```diff
@@ -238,7 +238,7 @@ object YarnSparkHadoopUtil {
     if (Utils.isWindows) {
       escapeForShell("-XX:OnOutOfMemoryError=taskkill /F /PID %%%%p")
     } else {
-      "-XX:OnOutOfMemoryError='kill %p'"
+      "-XX:OnOutOfMemoryError='echo OnOutOfMemoryError; kill %p'"
```

Comment:
Does this require `bash` to interpret, and do we know the JVM would execute the command in a  shell? if you're tested this and it works, OK